### ISSUE TITLE
When more than one extension method uses the same Func<object> to set a value, only the first updates the value after StateHasChanged is called.

### ISSUE THREAD

**Issue by cleissonalves** [AvaloniaUI/Avalonia.Markup.Declarative#40]
The following sample component shows the behavior. Can someone reproduce this results?

```csharp
public class Flags : ComponentBase
{
	public bool Flag { get; set; } = false;

	protected override object Build()
	{
		new StackPanel()
			.Ref(out var panel)
			.Children(
				new Button()
					.Content(() => Flag) //Updates
					.OnClick(_ => {Flag = !Flag; StateHasChanged();}),
				new Button()
					.Content("Buttom")
					.IsEnabled(() => Flag), //NEVER Updates
				new Label()
					.Content(() => Flag), //NEVER Updates
				new CheckBox()
					.IsChecked(() => Flag), //NEVER Updates
				new Label()
					.Content(() => $"Flag value is {Flag}"), //Updates
				new Label()
					.Content(() => $"Flag value is {Flag}"), //NEVER Updates
				new Label()
					.Content(() => $"Value is {Flag}"), //Updates
				new TextBlock()
					.Text(() => $"Value is {Flag}"), //NEVER Updates
				new Label()
					.Content(() => Convert.ToInt32(Flag)), //Updates
				new Label()
					.Content(() => Convert.ToInt32(Flag)) //NEVER Updates
			);

		return panel;
	}
}
```

**Comment by cleissonalves** [comment]
If i understood correctly, the `ControlPropertyExtension` calls the method `_set`, that adds a new `ViewPropertyComputedState` to the `ViewBase` `__viewComputedStates` HashSet.

Maybe, when the functions are the same, only the first is being added to this hashset. 

Is it because the way the `ViewPropertyComputedState` calculates the `GetHashCode` and the `Equals` method?

```csharp
internal abstract class ViewPropertyComputedState
{
    internal string? ExpressionString { get; set; }

    public override bool Equals(object? obj)
    {
        return obj is ViewPropertyComputedState state &&
               ExpressionString == state.ExpressionString;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(ExpressionString);
    }

    public abstract void OnPropertyChanged();
}
```

In these extension methods, the `expression` is `[CallerArgumentExpression("func")] string? expression = null`. 

```csharp
public static T Content<T>(
        this T control, 
        Func<Object> func, 
        Action<Object>? onChanged = null, 
        [CallerArgumentExpression("func")] string? expression = null) where T : ContentControl
   => control._set(ContentControl.ContentProperty, func, onChanged, expression);
```

Then if two functions are equal, the `string? expression` is the same, the `GetHashCode` and the `Equals` returns the same value. So only the first is added to the HashSet.

Does it make any sense? This is my only theory so far.

**Comment by cleissonalves** [comment]
To test the theory, i set the ExpressionString [on this line](https://github.com/AvaloniaUI/Avalonia.Markup.Declarative/blob/2f175e72ef17962c413f450d523ac6aa0cc5a975/src/Avalonia.Markup.Declarative/ViewPropertyComputedState.cs#L75) to a Guid: 

```csharp
//   ExpressionString = expressionString;
ExpressionString = Guid.NewGuid().ToString();
```

And now all the values on the sample component are updated when StateHasChanged is called.

**Comment by gritsenko** [comment]
@cleissonalves thanks for the research! Yes, expression is used as the key when keeping component state, for now... I have to think  a little to choose better implementation here

**Comment by gritsenko** [comment]
resolved in the latest nuget 11.1.3

**Comment by cleissonalves** [comment]
Nice. Everything is working now. Thank you!
